### PR TITLE
layout change UI

### LINKED ISSUE #594: Switch layouts from client
Switch back and forth between t-sne & umap, for instance

Thanks to @dpeerlab for the suggestion

### LINKED ISSUE #725: support loading arbitrary layouts
currently, `cellxgene launch` allows the user to select from the standard layouts that scanpy uses, each of which is stored in `.obsm` and prefaced with `X_`.

If a user has another layout (e.g. spatial), they cannot load it into cellxgene.

suggested change: allow the user to use any layout stored in an `.obsm` key with the syntax `X_<layout>`

preferred UX:

1. let user pass layout to use into CLI
2. if requested layout does not exist or no layout is passed in, inspect file and give user prompt with available layouts

### PR DESCRIPTION
**Pull request by bkmartinjr**
This PR completes the layout switch UI.  Changes:
* add tool bar menu to switch between available layouts
* add layout schema information to the REST API

Fixes #594 
Fixes #725

### CODE REVIEW

File: server/app/scanpy_engine/scanpy_engine.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def _create_schema(self):
                 "type": str(self.data.X.dtype),
             },
             "annotations": {"obs": [], "var": []},
+            "layout": {"obs": [], "var": []}
```
Comment (csweaver) [inline server/app/scanpy_engine/scanpy_engine.py]: With no plans on the horizon to support var layout, I'd just as soon leave it out.

File: server/app/scanpy_engine/scanpy_engine.py [hunk 2]
```diff
@@ -140,6 +141,14 @@ def _create_schema(self):
                     )
                 self.schema["annotations"][ax].append(ann_schema)
 
+        for layout in self.config['layout']:
+            ly_schema = {
```
Comment (csweaver) [inline server/app/scanpy_engine/scanpy_engine.py]: nit: `ly` is not how I would abbreviate layout (`lo` could be defensible, but I'd prefer `layout`)

File: server/app/scanpy_engine/scanpy_engine.py (hunk 1 above)
Comment (bkmartinjr) [reply server/app/scanpy_engine/scanpy_engine.py]: fixed

File: server/app/scanpy_engine/scanpy_engine.py (hunk 2 above)
Comment (bkmartinjr) [reply server/app/scanpy_engine/scanpy_engine.py]: fixed

**Pull request merged**